packages/service-library/src/servicelib/deferred_tasks/__init__.py

```diff
@@ -41,9 +43,10 @@
     ErrorResult --> FinishedWithError: gives up when out of retries or if cancelled
     Worker --> DeferredResult: success
 
-    DeferredResult --> °: calls [on_result]
-    FinishedWithError --> °°: calls [on_finished_with_error]
-    Worker --> °°°: task cancelled
+    DeferredResult --> (3): calls [on_result]
```

Comment:
Does a task gets reschedule (i.e. -->try again --> `SubmitTask`) every time it errors?

Reply:
If an event is stopped due to syste restart or other issues it will be requeued until it's processed correctly.
The task per se will also be retried in case it fails up to a number of max retries